app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
You are right about giving access to the `index` and `show` pages. But you are not following the requirements. The requirements say that you need to create necessary helper methods in the `ApplicationController`. This means that you need to create a custom method to enforce user log in and then use that in `before_action`. But you are using the `authenticate_user!` method which is coming from `devise`. Here is the requirement: [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)

- [ ] Please add the custom method to use in `before_action`.

Reply:
Thanks for the clarification! I've done it as required.